azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -186,9 +206,9 @@ def create_from_edge_environment(cls):
                 workload_uri=workload_uri,
                 api_version=api_version,
             )
-
-        pipeline = IoTHubPipeline(authentication_provider)
-        return cls(pipeline)
+        iothub_pipeline = pipeline.IoTHubPipeline(authentication_provider)
+        edge_pipeline = pipeline.EdgePipeline(authentication_provider)
```

Comment:
>edge_pipeline = pipeline.EdgePipeline(authentication_provider) [](start = 8, length = 62)

I'm glad I saw this.  One auth provider serving two pipelines has implications for sas renewal.  The auth provider needs to notify _both_ pipelines when the token is renewed.  No big deal, but important to get right.

Reply:
And, right now, the auth provider isn't notifying anyone on token renewal, hence issue #142, so you won't find any code to change yet.  

---
In reply to: [306955500](https://github.com/Azure/azure-iot-sdk-python-preview/pull/141#discussion_r306955500) [](ancestors = 306955500)